worker-workflow/src/test/java/com/github/cafdataprocessing/workflow/WorkflowManagerTest.java

```diff
@@ -100,5 +105,108 @@ public void noActionNameTest() throws WorkerException {
         }
 
     }
+    
+    @Test
+    public void onBeforeProcessDocumentTest() throws Exception
+    {
+        final Document document = getDocumentWithSubDocument();
+        document.getField("CAF_WORKFLOW_ACTION").add("lang_detect");
+
+        final WorkflowManager workflowManager = new WorkflowManager(document.getApplication(),
+                                     WorkflowDirectoryProvider.getWorkflowDirectory("workflow-manager-test"));
+
+        final Workflow workflow = workflowManager.get("test-workflow");
+        final String workflowScript = workflow.getWorkflowScript();
+        executeOnBeforeProcessDocumentScript(workflowScript, document);
+    }
 
+    private static void executeOnBeforeProcessDocumentScript(String workflowScript, Document document)
+    {
+        final ScriptEngine scriptEngine = new ScriptEngineManager().getEngineByName("nashorn");
+        final Invocable invocable = (Invocable) scriptEngine;
+        try {
+            scriptEngine.eval(workflowScript);
+            invokeOnBeforeFunction(invocable, document);
+        } catch (NoSuchMethodException | ScriptException ex) {
+            throw new RuntimeException(ex);
+        }
+
+    }
+
+    private static void invokeOnBeforeFunction(Invocable invocable, Document document) throws NoSuchMethodException, ScriptException
+    {
+        final CancelableDocumentEventObject cancelTaskEventObject = new CancelableDocumentEventObject(document);
+        boolean valueCheck = document.getField("CONTENT_PRIMARY").hasValues();
+        invocable.invokeFunction("onBeforeProcessDocument", cancelTaskEventObject);
+        System.out.println("e.cancel: " + cancelTaskEventObject.cancel);
+        assertEquals(!valueCheck, cancelTaskEventObject.cancel);
```

Comment:
Expected value should be a parameter to the method so you can test what happens when condition is met and not met.